File: back/src/main/java/br/com/feliva/back/models/Endereco.java
package br.com.feliva.back.models;

import br.com.feliva.sharedClass.db.Model;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name = "endereco")
public class Endereco extends Model<Integer> {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_endereco")
    private Integer idEndereco;

    @Column(length = 9)
    @NotNull(message = "Informe um cep.")
    private String cep;

    @NotNull(message = "Informe uma cidade.")
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id_cidade")
    private Cidade cidade;

    @NotNull(message = "Informe um bairro.")
    @Column(length = 50)
    private String bairro;

    @NotNull(message = "Informe um logradouro.")
    @Column(length = 50)
    private String logradouro;

    @NotNull(message = "Informe o numero do endereço.")
    @Column(length = 30)
    private String numero;

    @Column(length = 50)
    private String complemento;

    @Override
    public Integer getMMId() {
        return this.idEndereco;
    }
}

File: front/src/app/model/Endereco.ts
import {FormControl, FormGroup, Validators} from "@angular/forms";
import {Cidade} from "./Cidade";
import {Model} from "./Model";

export class Endereco extends Model{
  idEndereco?: number;
  cep?: string;
  cidade?:Cidade
  bairro?:string
  logradouro?:string
  numero?:string
  complemento?:string;

  constructor(entity?: Partial<Endereco>) {
    super(entity);
  }


  static CreateFormGroup(entity?:Endereco){
    console.log(entity?.cidade)
    return new FormGroup({
      idEndereco : new FormControl(entity?.idEndereco, []),
      cep : new FormControl(entity?.cep, [Validators.required]),
      estado : new FormControl(entity?.cidade?.estado, [Validators.required]),
      cidade : new FormControl(entity?.cidade, [Validators.required]),
      bairro : new FormControl(entity?.bairro, [Validators.required]),
      logradouro : new FormControl(entity?.logradouro, [Validators.required]),
      numero : new FormControl(entity?.numero, [Validators.required]),
      complemento : new FormControl(entity?.complemento, [Validators.required]),
    })
  }
}
